### ISSUE TITLE
Consider using mypyc

### ISSUE THREAD

**Issue by blink1073** [ipython/traitlets#864]
We might be able to use [mypyc](https://mypyc.readthedocs.io/en/latest/index.html) to offer accelerated versions of the package for specific architectures.

**Comment by bollwyvl** [comment]
Yep, as long as there's still a `-py3-none-any.whl` (as on [black](https://pypi.org/project/black/#files)), it _shouldn't_ be super disruptive for downstreams. All the same, such a change might  warrant a  _broad_ downstream canary testing before pulling the trigger... maybe a major version bump.

Of note, though, and maybe a bit late: it may be worth harmonizing some of the choices with [annotated-types](https://github.com/annotated-types/annotated-types), now used by `pydantic`, etc.
